src/rvsdg/optimize_direct_jumps.rs

```diff
@@ -0,0 +1,113 @@
+//! This module takes as input a [`SimpleCfgProgram`] and gets rid of any
+//! direct jumps between blocks.
+//! The block that is jumped to must have only one predecessor.
+//! This is used by `to_cfg` to clean up
+//! the output.
+
+use hashbrown::HashMap;
+use petgraph::{
+    stable_graph::{NodeIndex, StableDiGraph, StableGraph},
+    visit::Bfs,
+    visit::EdgeRef,
+    Direction,
+};
+
+use crate::cfg::{BasicBlock, Branch, Simple, SimpleCfgFunction, SimpleCfgProgram};
+
+struct JumpOptimizer<'a> {
+    simple_func: &'a SimpleCfgFunction,
+}
+
+impl<'a> JumpOptimizer<'a> {
+    fn optimized(&mut self) -> SimpleCfgFunction {
+        let mut resulting_graph: StableGraph<BasicBlock, Branch> = StableDiGraph::new();
+
+        // a map from nodes in the new graph to nodes in the
+        // old graph
+        // if a node was fused into another node, fix this
+        // so it points to the fused node
+        let mut node_mapping: HashMap<NodeIndex, NodeIndex> = HashMap::new();
+
+        let mut bfs = Bfs::new(&self.simple_func.graph, self.simple_func.entry);
```

Comment:
nit: My read here is that we need BFS here to ensure that node_mapping[&previous] always succeeds. Maybe mention that here?

Reply:
Yeah, good idea